src/Microsoft.ML.Api/TypedCursor.cs

```diff
@@ -292,42 +292,22 @@ private Action<TRow> GenerateSetter(IRow input, int index, InternalSchemaDefinit
                     else if (fieldType.GetElementType() == typeof(int))
                     {
                         Ch.Assert(colType.ItemType == NumberType.I4);
-                        return CreateConvertingVBufferSetter<DvInt4, int>(input, index, poke, peek, x => (int)x);
-                    }
-                    else if (fieldType.GetElementType() == typeof(int?))
-                    {
-                        Ch.Assert(colType.ItemType == NumberType.I4);
-                        return CreateConvertingVBufferSetter<DvInt4, int?>(input, index, poke, peek, x => (int?)x);
+                        return CreateConvertingVBufferSetter<int, int>(input, index, poke, peek, x => (int)x);
```

Comment:
>return CreateConvertingVBufferSetter<int, int>(input, index, poke, peek, x => (int)x); [](start = 24, length = 86)

These were previously necessary when the types are different. Why are they necessary now? For example I definitely do not see things for `float` or `double` in this list.

Reply:
agreed, this was a miss. No need for this conversion. 

---
In reply to: [212031778](https://github.com/dotnet/machinelearning/pull/683#discussion_r212031778) [](ancestors = 212031778)